frontend/packages/console-shared/src/hooks/fullscreen.ts

```diff
@@ -0,0 +1,43 @@
+import * as React from 'react';
+
+export const useFullscreen = <T extends HTMLElement>(): [
+  boolean,
+  (node: T) => void,
+  () => void,
+] => {
+  const [isFullscreen, setIsFullscreen] = React.useState<boolean>(false);
+  const fullscreenRef = React.useRef<boolean>(isFullscreen);
+  fullscreenRef.current = isFullscreen;
+  const elementRef = React.useRef<T>();
+
+  const listener = (event) => {
+    setIsFullscreen(document.fullscreenElement === event.target);
+  };
+
+  const targetCallbackRef = React.useCallback((node: T) => {
+    if (node != null) {
+      elementRef.current = node;
+      if (document.fullscreenEnabled) {
+        node.addEventListener('fullscreenchange', listener);
+        node.addEventListener('fullscreenerror', listener);
+      }
+    }
```

Comment:
You should call `removeEventListener` when the incoming node is changing or is set to null.